### PR TITLE
added task_01, task_02 for review

### PR DESCRIPTION
**Pull request by elenaalkoli**

### CODE REVIEW

File: hw_08/task_02.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+/*
+7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+Пример:
+deleteEnterprise(1)
+*/
+console.log("-------------------task_7-------------------")
+
+function deleteEnterprise(id) {
+  const newEnterprises = enterprises.filter((object) => object.id !== id); //новый массив
```
Comment (anatoly-karpovich) [inline hw_08/task_02.js:240]: You did not change the actual Enterprises array. Need to use splice to delete actual enterprise from array

File: hw_08/task_02.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+/*
+8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+Пример:
+deleteDepartment(3)
+*/
+console.log("-------------------task_8-------------------")
+
+function deleteDepartment(id) {
+  for (let object of enterprises) {
+    const newStructure = object.departments.filter(enterprise => enterprise.id !== id && enterprise.employees_count !== 0);
```
Comment (anatoly-karpovich) [inline hw_08/task_02.js:254]: same, find enterprise and using splice delete the nested department by its index

File: hw_08/task_02.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+Пример:
+moveEmployees(2, 3)
+*/
+console.log("-------------------task_9-------------------")
+
+function moveEmployees(idFrom, idTo) {
+  const departmentFrom = getDepartmentById(idFrom); //отдел откуда переносим employees_count
+  const departmentTo = getDepartmentById(idTo); //отдел куда 
+  departmentTo.employees_count += departmentFrom.employees_count;
```
Comment (anatoly-karpovich) [inline hw_08/task_02.js:273]: Need to check that this departments are from one enterprise) not obligatory, but we have this requirement)

**Review by anatoly-karpovich — APPROVED** [review]

File: hw_08/task_02.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+Пример:
+addEnterprise("Название нового предприятия")
+*/
+console.log("-------------------task_3-------------------")
+
+function addNewEnterprise(name) {
+  let newDepartment = {
+    id: getLastId(),
+    name,
+    departments: null
```
Comment (VladimirLQA) [inline hw_08/task_02.js:134]:
```suggestion
    departments: []
```

File: hw_08/task_02.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+    console.log(`No department found by id: ${id}`);
+  }
+  return foundedDepartment;
+}
+
+function createNewDepartmentByName(name) { //создание нового департамента
+  const newDepartment = {
+    id: getLastId(),
+    name,
+    employees_count: null,
```
Comment (VladimirLQA) [inline hw_08/task_02.js:184]:
```suggestion
    employees_count: 0,
```

File: hw_08/task_02.js [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+Пример:
+editEnterprise(1, "Новое название предприятия")
+*/
+console.log("-------------------task_5-------------------")
+
+function editDepartmentName(id, newDepName) {
+  foundedDepartment = enterprises.find((object) => object.id = id);
+  foundedDepartment.name = newDepName;
+  return foundedDepartment;
```
Comment (VladimirLQA) [inline hw_08/task_02.js:201]: No need to return any value
```suggestion
  const foundedDepartment = enterprises.find((object) => object.id === id);
  foundedDepartment.name = newDepName;
```

File: hw_08/task_02.js [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+console.log(addNewDepartment(5, "Elena's department"))
+
+/*
+5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+Пример:
+editEnterprise(1, "Новое название предприятия")
+*/
+console.log("-------------------task_5-------------------")
+
+function editDepartmentName(id, newDepName) {
```
Comment (VladimirLQA) [inline hw_08/task_02.js:198]: department or enterprise ?
```suggestion
function editDepartmentName(id, newDepName) {
```

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**